core/src/main/java/cucumber/runner/SynchronizedEventBus.java

```diff
@@ -1,41 +1,39 @@
 package cucumber.runner;
 
 import cucumber.api.event.Event;
-import cucumber.api.event.EventHandler;
 
-import java.util.Collections;
+import java.util.Collection;
 
-public class SynchronizedEventBus implements EventBus {
+public final class SynchronizedEventBus extends AbstractEventBus {
 
     private final EventBus delegate;
 
-    public SynchronizedEventBus(final EventBus delegate) {
+    public static SynchronizedEventBus synchronize(EventBus eventBus) {
+        if (eventBus instanceof SynchronizedEventBus) {
+            return (SynchronizedEventBus) eventBus;
+        }
+
+        return new SynchronizedEventBus(eventBus);
```

Comment:
This will not work, you cannot have multiple instances of the class and just use the `synchronized` keyword on the method.  The keyword on the method only effects the instance, not all instances - this is why I used a static sync object before

Example
```
public static void main(String[] args) throws InterruptedException {
    final ExecutorService executorService = Executors.newFixedThreadPool(2);
    executorService.submit(new SynMethod());
    Thread.sleep(100L);
    executorService.submit(new SynMethod());
}
private static class SynMethod implements Runnable {
    synchronized void doSomething() {
        DateFormat df = new SimpleDateFormat("hh:MM:ss.SSS");
        System.out.println(Thread.currentThread().getId() + " - hey there " + df.format(new Date()))
        try {
            Thread.sleep(3000L);
        }
        catch (InterruptedException e) {
            e.printStackTrace();
        }
        System.out.println(Thread.currentThread().getId() + " - all done " + df.format(new Date()));
    }
    @Override
    public void run() {
        doSomething();
    }
}
```
```
11 - hey there 09:06:11.221
12 - hey there 09:06:11.298
11 - all done 09:06:14.227
12 - all done 09:06:14.298
```